### PR TITLE
Implement string method `to_titlecase()`

### PR DESCRIPTION
**Pull request by etiennebacher**

### CODE REVIEW

**Comment by etiennebacher** [comment]
This only succeeds with full-features enabled, maybe because this function is marked as `nightly`?

https://github.com/pola-rs/polars/blob/0d90a06536804e7c041d72a244f0dcb4e104c66b/crates/polars-plan/src/dsl/string.rs#L488

**Comment by sorhawell** [comment]
Uhh have seen such method that before relying on nightly I think.
I added some conditional compilation. Now `$to_titlecase()`will throw an error if not compiled with "full_feaures" aka envvar "RPOLARS_FULL_FEATURES". There is a function to check on R side if full features also `polars:::polars_has_full_feaures()`

File: src/rust/src/rlib.rs [hunk 1]
```diff
@@ -265,6 +265,17 @@ fn test_wrong_call_pl_lit(robj: Robj) -> RResult<Robj> {
...
+#[cfg(feature = "full_features")]
+const POLARS_FEATURE_FULL_FEATURES: bool = true;
+
+#[cfg(not(feature = "full_features"))]
+const POLARS_FEATURE_FULL_FEATURES: bool = false;
+
+#[extendr]
+fn polars_has_full_features() -> bool {
+    POLARS_FEATURE_FULL_FEATURES
+}
```
Comment (eitsupi) [inline src/rust/src/rlib.rs:277]: How about use the `!cfg` macro?

File: src/rust/src/lazy/dsl.rs [hunk 2]
```diff
@@ -2377,6 +2381,22 @@ impl Expr {
...
+    Ok(expr.0.clone().str().to_titlecase().into())
+}
+
+#[cfg(not(feature = "full_features"))]
+fn f_str_to_titlecase(_expr: &Expr) -> RResult<Expr> {
+    rerr().plain(
+        "$to_titlecase() is only available with 'full_features' enabled. Try our github \
+    binary releases or compile with env var RPOLARS_FULL_FEATURES = 'true'",
+    )
+}
```
Comment (eitsupi) [inline src/rust/src/lazy/dsl.rs:2398]: Can't we evaluate the feature flag in the function?

**Review by eitsupi — CHANGES_REQUESTED** [review]

File: R/extendr-wrappers.R [hunk 3]
```diff
@@ -73,6 +73,8 @@ test_robj_to_expr <- function(robj) .Call(wrap__test_robj_to_expr, robj)
 
 test_wrong_call_pl_lit <- function(robj) .Call(wrap__test_wrong_call_pl_lit, robj)
 
+polars_features <- function() .Call(wrap__polars_features)
```
Comment (etiennebacher) [inline R/extendr-wrappers.R:76]: This is a bit a duplicate of `polars::pl$polars_info()`, is it possible to merge the two?
```r
polars::pl$polars_info()
#> r-polars package version : 0.8.0.9000
#> rust-polars crate version: 0.32.0
#> 
#> Features:          
#> simd FALSE

polars_features()
#> $full_features
#> [1] FALSE
#> 
#> $default
#> [1] TRUE
#> 
#> $simd
#> [1] FALSE
#> 
#> $rpolars_debug_print
#> [1] FALSE
```

**Review by etiennebacher — COMMENTED** [review]
thanks for fixing it @sorhawell, I have a comment about `polars_features()`

File: NEWS.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
 # polars (development version)
 
+## What's changed
+
+- New string method `to_titlecase()` (#371).
+
+
```
Comment (eitsupi) [inline NEWS.md:7]:
```suggestion
```

File: tests/testthat/test-expr_string.R [hunk 5]
```diff
@@ -126,7 +126,9 @@ test_that("str$concat", {
 })
 
 
-test_that("str$to_uppercase to_lowercase", {
+test_that("to_uppercase, to_lowercase", {
+
+
```
Comment (eitsupi) [inline tests/testthat/test-expr_string.R:131]:
```suggestion
```

File: tests/testthat/test-expr_string.R [hunk 6]
```diff
@@ -139,6 +141,22 @@ test_that("str$to_uppercase to_lowercase", {
     df$select(pl$col("foo")$str$to_lowercase())$to_list()$foo,
     tolower(df$to_list()$foo)
   )
+
+})
+
```
Comment (eitsupi) [inline tests/testthat/test-expr_string.R:146]:
```suggestion
```

**Review by eitsupi — APPROVED** [review]
Looks good, thanks!

File: src/rust/src/rlib.rs [hunk 7]
```diff
@@ -265,6 +265,16 @@ fn test_wrong_call_pl_lit(robj: Robj) -> RResult<Robj> {
...
 
+#[extendr]
+fn polars_features() -> List {
+    list!(
+        full_features = cfg!(feature = "full_features"),
+        default = cfg!(feature = "default"),
+        simd = cfg!(feature = "simd"),
+        rpolars_debug_print = cfg!(feature = "rpolars_debug_print")
+    )
+}
```
Comment (eitsupi) [inline src/rust/src/rlib.rs:276]: This seems unused.

**Pull request merged**